### DISCUSSION TITLE
如何动态给某些静态库添加连接命令？

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#2922] [Q&A]
某些特定库需要添加whole-archive 既 -Wl,--whole-archive liba.a -Wl,-no-whole-archive
xmake中
target("a")
    set_kind("static")
    set_targetdir("$(buildir)/lib")
    add_includedirs("inc",{public = true})
    add_files("src/*.c",{public = true})
target_end()
target("app.elf")
    set_kind("binary")
    add_deps(a)
使用add_deps(a)后如何添加呢？

**Comment by waruqi** [comment]
暂时不支持 https://github.com/xmake-io/xmake/issues/1452

你只能禁用 link 继承，add_deps("a", {inherit = false})

然后 add_linkdirs add_ldflags 自己加
